AdyenCard/Components/GiftCardComponent/GiftCardComponent.swift

```diff
@@ -62,6 +39,20 @@ public final class GiftCardComponent: PartialPaymentComponent,
     /// The delegate that handles shopper confirmation UI when the balance of the gift card is sufficient to pay.
     public weak var readyToSubmitComponentDelegate: ReadyToSubmitPaymentComponentDelegate?
 
+    /// :nodoc:
+    public var clientKey: String? {
```

Comment:
GiftCardComponent doesn't need `clientKey` on it own - it pass it to `cardPublicKeyProvider`; and it is set on `init`.

We can simply do 
```swift
 public var clientKey: String? { cardPublicKeyProvider.clientKey }
```

---

Maybe it is easier to just set both `clientKey` and `environment` in `init` as we did with DropInComponent?

Reply:
this is out of scope, will change this later